r-package/grf/vignettes/rate_cv.Rmd

```diff
@@ -0,0 +1,438 @@
+  nuisance.forest = causal_forest(X, Y, W)
+  DR.scores = get_scores(nuisance.forest)
+
+  for (k in 1:num.folds) {
+    train = unlist(samples.by.fold[-k])
+    test = samples.by.fold[[k]]
+
+    cate.forest = causal_forest(X[train, ], Y[train], W[train])
+
+    cate.hat.test = predict(cate.forest, X[test, ])$predictions
+
+    rate.fold = rank_average_treatment_effect.fit(DR.scores[test], cate.hat.test)
+    t.statistics = c(t.statistics, rate.fold$estimate / rate.fold$std.err)
+  }
+
+  p.value.naive = 2 * pnorm(-abs(sum(t.statistics) / sqrt(num.folds)))
+
+  p.value.naive
+}
+```
+Next, we simulate synthetic data with no treatment heterogeneity, and form a Monte Carlo estimate of our naive procedure's rejection probability, using a 5% confidence level:
+
+```{r, eval=FALSE}
+res = t(replicate(500, {
+  n = 1500
+  p = 5
+  X = matrix(rnorm(n * p), n, p)
+  W = rbinom(n, 1, 0.5)
+  Y = pmin(X[, 3], 0) + rnorm(n)
+
+  pval.naive = rate_cv_naive(X, Y, W)
+
+  c(
+    reject = pval.naive <= 0.05
+  )
+}))
+paste("Rejection rate:", mean(res) * 100, "%")
+#> [1] "Rejection rate: 16.8 %"
+```
+At a 5% level, any test with a rejection rate above 5% is an invalid test.
+
+## Sequential cross-fold estimation
+We can improve on the above by considering another form of cross-fold estimation. In **sequential** cross-fold estimation we break the independence structure between folds by only using training folds in sequential order. As illustrated in the figure below, this strategy will give us $K - 1$ separate $t$-values.
+
+```{r, echo=FALSE}
+plot(NULL, xlim = c(1, 6), ylim = c(1, 6), xaxt = 'n', yaxt = 'n',
+     xlab = "Fold ID",
+     ylab = "Estimation step",
+     main = "Sequential 5-fold estimation"
+)
+
+lines(c(1, 2), c(4,4), lwd = 4, col = "blue")
+lines(c(2, 3), c(4,4), lwd = 4, col = "red")
+text(1.5, 4.25, offset = 0, "1", col = "blue")
+text(2.5, 4.25, offset = 0, "2", col = "red")
+
+lines(c(1, 3), c(3,3), lwd = 4, col = "blue")
+lines(c(3, 4), c(3,3), lwd = 4, col = "red")
+text(1.5, 3.25, offset = 0, "1", col = "blue")
+text(2.5, 3.25, offset = 0, "2", col = "blue")
+text(3.5, 3.25, offset = 0, "3", col = "red")
+
+lines(c(1, 4), c(2,2), lwd = 4, col = "blue")
+lines(c(4, 5), c(2,2), lwd = 4, col = "red")
+text(1.5, 2.25, offset = 0, "1", col = "blue")
+text(2.5, 2.25, offset = 0, "2", col = "blue")
+text(3.5, 2.25, offset = 0, "3", col = "blue")
+text(4.5, 2.25, offset = 0, "4", col = "red")
+
+lines(c(1, 5), c(1,1), lwd = 4, col = "blue")
+lines(c(5, 6), c(1,1), lwd = 4, col = "red")
+text(1.5, 1.25, offset = 0, "1", col = "blue")
+text(2.5, 1.25, offset = 0, "2", col = "blue")
+text(3.5, 1.25, offset = 0, "3", col = "blue")
+text(4.5, 1.25, offset = 0, "4", col = "blue")
+text(5.5, 1.25, offset = 0, "5", col = "red")
+
+text(3, 5.5, "Train", col = "blue")
+text(4, 5.5, "Test", col = "red")
+```
+
+This construction is going to make our $t$-values satisfy a martingale property and allow for valid aggregation. The steps are:
```

Comment:
Add citations to LvdL and the sequential validation comment as references for the martingale property claim